assignment4/IslandCounterTest.java

```diff
@@ -0,0 +1,35 @@
+package com.katermar.assignment4;
+
+import java.util.Arrays;
+
+/**
+ * Created by katermar on 6/21/2017.
+ */
+public class IslandCounterTest {
+
+    public static void main(String args[]) {
+        char map[][] = {{'F', 'T', 'F', 'T'},
+                        {'T', 'T', 'F', 'F'},
+                        {'F', 'F', 'T', 'F'},
+                        {'F', 'F', 'T', 'F'}};
+        char emptyArray[][] = {{}, {}};
+        testIslandCounter(3, map);
+        testIslandCounter(2, map);
+        testIslandCounter(0, emptyArray);
+    }
+
+    public static void testIslandCounter(int expectedResult, char map[][]) {
+        char checkEmpty[][] = new char[map.length][map[0].length];
+        if (Arrays.deepEquals(map, checkEmpty)) {
+            System.out.println("ERROR! Array is EMPTY! Test FAILED!");

```

Comment:
Why? Assignment does not forbid empty array. Also the array you use as empty is not really empty. It is one of edge conditions, but there is another one: `{}`.

Regarding checking if array is empty, why not just check lengths rather than comparing with an object?